dev/Scroller/Scroller.cpp

```diff
@@ -1312,16 +1312,35 @@ winrt::float2 Scroller::ComputePositionFromOffsets(double zoomedHorizontalOffset
     return winrt::float2(static_cast<float>(zoomedHorizontalOffset + minPosition.x), static_cast<float>(zoomedVerticalOffset + minPosition.y));
 }
 
-// Evaluate what the value will be once the snap points have been applied.
-double Scroller::ComputeValueAfterSnapPoints(double value, const std::set<winrt::ScrollerSnapPointBase, winrtProjectionComparator>& snapPoints)
+// Evaluate what the value will be once the scroll snap points have been applied.
+double Scroller::ComputeValueAfterScrollSnapPoints(
+    double value,
+    const std::set<winrt::ScrollSnapPointBase, winrtProjectionComparator>& snapPointsSet)
 {
-    for (winrt::ScrollerSnapPointBase winrtSnapPoint : snapPoints)
+    for (winrt::ScrollSnapPointBase winrtSnapPoint : snapPointsSet)
     {
-        auto snapPoint = winrt::get_self<ScrollerSnapPointBase>(winrtSnapPoint);
+        auto snapPoint = winrt::get_self<ScrollSnapPointBase>(winrtSnapPoint);
         if (std::get<0>(snapPoint->ActualApplicableZone()) <= value &&
             std::get<1>(snapPoint->ActualApplicableZone()) >= value)
         {
-            return ((double)snapPoint->Evaluate((float)value));
+            return (static_cast<double>(snapPoint->Evaluate(static_cast<float>(value))));
+        }
+    }
+    return value;
+}
+
+// Evaluate what the value will be once the zoom snap points have been applied.
+double Scroller::ComputeValueAfterZoomSnapPoints(
```

Comment:
>ComputeValueAfterZoomSnapPoints [](start = 17, length = 31)

This code is equivalent to the above method except for they type of the snap points. Can you combine these into a single Generic function?

Reply:
Using templated methods is elegant indeed. Thanks.
Done.